node_modules/col-activities/lib/api.js

```diff
@@ -178,26 +178,29 @@ var exportActivities = module.exports.exportActivities = function(ctx, callback)
  * @param  {String}           objectType                          The type of the object on which the activity is taking place. One of `CollabosphereConstants.ACTIVITY.OBJECT_TYPES`
  * @param  {Object}           [metadata]                          Additional metadata that is associated with the activity. For example, when creating a submission activity it might be useful to store the submission id, the attempt number and the attachment ids
  * @param  {User}             [actor]                             The user performing the activity when different than the user earning activity points
+ * @param  {Number}           [assetId]                           The associated asset id when different from the object id (e.g., for comments)
  * @param  {Function}         callback                            Standard callback function
  * @param  {Object}           callback.err                        An error that occurred, if any
  * @param  {Activity}         callback.activity                   The created activity
  */
-var createActivity = module.exports.createActivity = function(course, user, type, objectId, objectType, metadata, actor, callback) {
+var createActivity = module.exports.createActivity = function(course, user, type, objectId, objectType, metadata, actor, assetId, callback) {
```

Comment:
@nicolaasmatthijs If we don't want to change the function signature, I'm fine with passing the assetId in as part of the metadata; but in that case we'd still want to extract it within the function and store it in its own column to enable SQL joins. Make sense?